app/helpers/mb_form_builder.rb

```diff
@@ -21,11 +22,27 @@ def mb_input_field(
       autocapitalize: "off",
       spellcheck: "false",
     }.merge(options)
+
+    if object.respond_to?(:hash_key_attribute?) &&
```

Comment:
Ben and I found this line particularly confusing! The tests help provide some clarity, but we still had a problem parsing and fully understanding what is going on. Can you think of any way of making the intentions more obvious?